src/poetry/console/commands/build.py

```diff
@@ -114,13 +114,22 @@ def _get_builder(self) -> Callable[..., None]:
 
         return self._build
 
+    def _has_build_backend_defined(self) -> bool:
+        return "build-backend" in self.poetry.file.read().get("build-system", {})
```

Comment:
Would be good to move this to be something like `self.poetry.pytpoject.build_system.is_fallback`.

But that will need an additional property in https://github.com/python-poetry/poetry-core/blob/d97577e664d59173482a78f70e8f743cc5468851/src/poetry/core/pyproject/tables.py#L12

But for now something like this should work. 

```suggestion
        return "build-backend" in self.poetry.pyproject.data.get("build-system", {})
```

Reply:
Ah, I wasn't aware of the `data` attribute of `pyproject` :+1: 

I thought about a flag as well. But I don't wanted to change `poetry-core` before. And we would pollute again `poetry-core` with something we only need in `poetry`. Actually we only need it for `poetry build`.